mesonbuild/backend/ninjabackend.py

```diff
@@ -243,10 +244,12 @@ def generate(self, interp):
 
     # http://clang.llvm.org/docs/JSONCompilationDatabase.html
     def generate_compdb(self):
-        pch_compilers = ['%s_PCH' % i for i in self.build.compilers]
-        native_compilers = ['%s_COMPILER' % i for i in self.build.compilers]
-        cross_compilers = ['%s_CROSS_COMPILER' % i for i in self.build.cross_compilers]
-        ninja_compdb = [self.ninja_command, '-t', 'compdb'] + pch_compilers + native_compilers + cross_compilers
+        rules = [rule
+                 for for_machine in list(MachineChoice)
+                 for lang in self.build.compilers[for_machine]
+                 for rule in [self.get_compiler_rule_name(lang, for_machine),
+                              self.get_pch_rule_name(lang, for_machine)]]
```

Comment:
We now support precompiled headers with cross too!

Reply:
this really is a case not to use a comprehension and use nested for loops. This is mind bending to read.